x-pack/plugins/security_solution/public/explore/network/components/embeddables/embedded_map.tsx

```diff
@@ -119,31 +120,55 @@ export const EmbeddedMapComponent = ({
   const { kibanaDataViews } = useDeepEqualSelector((state) => getDataViewsSelector(state));
   const { selectedPatterns } = useSourcererDataView(SourcererScopeName.default);
 
-  const [mapIndexPatterns, setMapIndexPatterns] = useState(
-    kibanaDataViews.filter((dataView) => selectedPatterns.includes(dataView.title))
-  );
+  const isFieldInIndexPattern = useIsFieldInIndexPattern();
+
+  const [mapDataViews, setMapDataViews] = useState<SourcererDataView[]>([]);
+
+  const availableDataViews = useMemo(() => {
+    const dataViews = kibanaDataViews.filter((dataView) =>
+      selectedPatterns.includes(dataView.title)
+    );
+    if (selectedPatterns.length > 0 && dataViews.length === 0) {
+      setIsIndexError(true);
+    }
+    return dataViews;
+  }, [kibanaDataViews, selectedPatterns]);
+
+  useEffect(() => {
+    let canceled = false;
+
+    const fetchData = async () => {
+      try {
+        const apiResponse = await Promise.all(
+          availableDataViews.map(async ({ title }) =>
+            isFieldInIndexPattern(title, getRequiredMapsFields(title))
+          )
+        );
+        // ensures only index patterns with maps fields are passed
+        const goodDataViews = availableDataViews.filter((_, i) => apiResponse[i] ?? false);
+        if (!canceled) {
+          setMapDataViews(goodDataViews);
+        }
+      } catch {
```

Comment:
I think that we should not swallow errors like that. It could conceal some implementation mistakes. Let's display an error message to the user like`toasts.addError`.